diff --git a/src/wg-quick/linux.bash b/src/wg-quick/linux.bash
index 40d6410..9ebbd30 100755
--- a/src/wg-quick/linux.bash
+++ b/src/wg-quick/linux.bash
@@ -87,6 +87,10 @@ auto_su() {
 
 add_if() {
 	local ret
+	if [[ ${AWG_QUICK_FORCE_USERSPACE:-0} == 1 ]]; then
+		cmd "${WG_QUICK_USERSPACE_IMPLEMENTATION:-amneziawg-go}" "$INTERFACE"
+		return
+	fi
 	if ! cmd ip link add "$INTERFACE" type amneziawg; then
 		ret=$?
 		[[ -e /sys/module/amneziawg ]] || ! command -v "${WG_QUICK_USERSPACE_IMPLEMENTATION:-amneziawg-go}" >/dev/null && exit $ret
